org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/ClassfileContentHandlerTest.java

```diff
@@ -30,20 +34,26 @@
 
 	private IProject project;
 	private ClassfileContentHandler handler;
+	private PreferenceManager preferenceManager;
+	private Preferences preferences;
 
 	@Before
 	public void setup() throws Exception {
 		importProjects("maven/salut");
 		project = WorkspaceHelper.getProject("salut");
-		handler = new ClassfileContentHandler();
+		preferenceManager = mock(PreferenceManager.class);
+		preferences = mock(Preferences.class);
+		when(preferences.getDecompilerId()).thenReturn("none");
```

Comment:
by default, it should return "default" (i.e disassembler). "none" implies no disassembly/decompilation will be attempted whatsoever (Which could be a possibility)